qa/L0_shared_memory/shared_memory_test.py

```diff
@@ -29,124 +29,143 @@
+        self._protocol = protocol
+        self._test_passed = False
+        self._original_stdout, self._original_stderr = None, None
+        self._log_file_path = log_file_path
+
+        if self._protocol == "http":
+            self._url = self._tritonserver_ipaddr + ":8000"
+            self._triton_client = httpclient.InferenceServerClient(
+                self._url, verbose=True
+            )
         else:
-            triton_client = grpcclient.InferenceServerClient(_url, verbose=True)
-        shm_op0_handle = shm.create_shared_memory_region("dummy_data", "/dummy_data", 8)
-        shm.set_shared_memory_region(
-            shm_op0_handle, [np.array([1, 2], dtype=np.float32)]
+            self._url = self._tritonserver_ipaddr + ":8001"
+            self._triton_client = grpcclient.InferenceServerClient(
+                self._url, verbose=True
+            )
+        self._build_model_repo()
+        self._build_server_args()
+        self._shared_memory_test_server_log = open(log_file_path, "w")
+        self._server_process = util.run_server(
+            server_executable=self._server_executable,
+            launch_command=self._launch_command,
+            log_file=self._shared_memory_test_server_log,
+        )
+        util.wait_for_server_ready(
+            self._server_process, self._triton_client, self._timeout
+        )
+        # Redirect console output to client log and begin test
+        self._client_log = open(SharedMemoryTest.shared_memory_test_client_log, "a")
+        self._original_stdout, self._original_stderr = util.stream_to_log(
+            self._client_log
         )
-        triton_client.register_system_shared_memory("dummy_data", "/dummy_data", 8)
-        shm_status = triton_client.get_system_shared_memory_status()
-        if _protocol == "http":
-            self.assertTrue(len(shm_status) == 1)
-        else:
-            self.assertTrue(len(shm_status.regions) == 1)
-        shm.destroy_shared_memory_region(shm_op0_handle)
 
-    def test_unregister_before_register(self):
-        # Create a valid system shared memory region and unregister before register
-        if _protocol == "http":
-            triton_client = httpclient.InferenceServerClient(_url, verbose=True)
-        else:
-            triton_client = grpcclient.InferenceServerClient(_url, verbose=True)
-        shm_op0_handle = shm.create_shared_memory_region("dummy_data", "/dummy_data", 8)
-        triton_client.unregister_system_shared_memory("dummy_data")
-        shm_status = triton_client.get_system_shared_memory_status()
-        if _protocol == "http":
-            self.assertTrue(len(shm_status) == 0)
+    def _build_server_args(self):
+        if self._test_windows:
+            backend_dir = "C:\\opt\\tritonserver\\backends"
+            model_dir = "C:\\opt\\tritonserver\\qa\\L0_shared_memory\\models"
+            self._server_executable = "C:\\opt\\tritonserver\\bin\\tritonserver.exe"
```

Comment:
Probably more of a random note or follow-up, but I was under the impression something like `Pathlib.Path("/opt/tritonserver/backends")` would translate to `"C:\\opt\\tritonserver\\backends"` for free when run on Windows. If so you could probably condense the cases to work for both.

Did you see otherwise?

Reply:
No, I believe you are right. ATM they are set to my local path and were hardcoded for convenience. They need to be modified and will once I determine the CI environment.